roles/cifmw_cephadm/tasks/configure_object.yml

```diff
@@ -38,21 +38,97 @@
       oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role create swiftoperator
       oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role create ResellerAdmin
-      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user swift --project service member
-      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --user swift --project service admin
-      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack endpoint create --region regionOne object-store public http://{{ cifmw_cephadm_vip }}:8080/swift/v1/AUTH_%\(tenant_id\)s
-      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack endpoint create --region regionOne object-store internal http://{{ cifmw_cephadm_vip }}:8080/swift/v1/AUTH_%\(tenant_id\)s
-      oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role add --project admin --user admin swiftoperator
+  delegate_to: localhost
+  when:
+    - cifmw_openshift_kubeconfig is defined
+    - cifmw_ceph_daemons_layout.rgw_enabled | default(true) | bool
+    - not swift_in_ctlplane.stdout | bool
+    - swift_endpoints_count.stdout == "0"
+
+- name: Capture required UUIDs of swift user/service
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+  delegate_to: localhost
+  when: cifmw_openshift_kubeconfig is defined
+  block:
+    - name: Get uuid for user 'swift'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user show swift -c id -f value"
+      register: user_swift_uuid
+
+    - name: Get uuid for service 'swift'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack service show swift -c id -f value"
+      register: service_swift_uuid
+
+    - name: Get uuid for role 'member'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show member -c id -f value"
+      register: role_member_uuid
+
+    - name: Get uuid for role 'admin'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show admin -c id -f value"
+      register: role_admin_uuid
+
+    - name: Get uuid for project 'admin'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack project show admin -c id -f value"
+      register: project_admin_uuid
+
+    - name: Get uuid for user 'admin'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack user show admin -c id -f value"
+      register: user_admin_uuid
+
+    - name: Get uuid for role 'swiftoperator'
+      ansible.builtin.shell: "set -o pipefail && oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack role show swiftoperator -c id -f value"
+      register: role_swiftoperator_uuid
+
```

Comment:
basically you are executing the same command for different users.

I'd loop on users and work on the resulting registered list to obtain those values.

WDYT?

Reply:
@rebtoor Thanks for your review but they are not same commands.
We capture not only different users uuid but also roles, projects, services. 
so i prefer them to be separate commands.

Reply:
it doesn't change too much TBH.

You can use a list of hashes or a dict.

e.g.
```
- name: Get required uuids
  ansible.builtin.command: "oc -n {{ cifmw_cephadm_ns }} rsh openstackclient openstack {{ os_command }} show {{ os_command_object }} -c id -f value"
  register: user_uuid
  loop:
    - { os_command: 'service', os_command_object: 'swift' }
    - { os_command: 'role', os_command_object: 'member' }
    - { os_command: 'role', os_command_object: 'admin' }
    - { os_command: 'project', os_command_object: 'admin' }
    - { os_command: 'user', os_command_object: 'admin' }
    - { os_command: 'role', os_command_object: 'swiftoperator' }
```

then you can can use `user_uuid.results` list to obtain the stdout you need.

Reply:
@rebtoor thanks, i updated the code.